File: php/features.php
<?php
declare(strict_types=1);

// Coingecko SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class CoingeckoFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new CoingeckoBaseFeature();
            case "test":
                return new CoingeckoTestFeature();
            default:
                return new CoingeckoBaseFeature();
        }
    }
}

File: py/features.py
# Coingecko SDK feature factory

from feature.base_feature import CoingeckoBaseFeature
from feature.test_feature import CoingeckoTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CoingeckoBaseFeature(),
        "test": lambda: CoingeckoTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
